implementations/rust/ockam/ockam_identity/src/secure_channel/encryptor_worker.rs

```diff
@@ -141,16 +172,46 @@ impl EncryptorWorker {
         // Remove our address
         let _ = onward_route.step();
 
+        let payload = msg.into_payload();
         let msg = PlaintextPayloadMessage {
             onward_route,
             return_route,
-            payload: msg.into_local_message().into_payload(),
+            payload: &payload,
         };
         let msg = SecureChannelMessage::Payload(msg);
 
-        let msg = self.encrypt(ctx, msg).await?;
+        let payload = {
+            // This is a workaround to keep backcompatibility with an extra
+            // bare encoding of the raw message (Vec<u8>).
+            // In bare, a variable length array is simply represented by a
+            // variable length integer followed by the raw bytes.
+            // The idea is first to calculate the size of the encrypted payload,
+            // so we can calculate the size of the variable length integer.
+            // The goal is to prepend the variable length integer to the buffer
+            // before it's actually written, so we can write the whole encrypted
+            // payload without any extra copies.
+
+            let encoded_payload = minicbor::to_vec(&msg)?;
+            // we assume this calculation is exact
+            let encrypted_payload_size = 8 + 8 + 8 + encoded_payload.len();
```

Comment:
It's 8 bytes nonce + 16 bytes auth tag. The result is correct but let's create corresponding public constants in the `ockam_vault` crate